src/Components/PropertyInspector/StandalonePropertyInspector.state.ts

```diff
@@ -234,6 +252,115 @@ const StandalonePropertyInspectorReducer = produce(
+                const targetNode = PropertyInspectorModel.findPropertyTreeNodeRefRecursively(
+                    draft.propertyTreeNodes,
+                    arrayNode.path
+                );
+
+                if (Array.isArray(targetNode.children)) {
+                    targetNode.children = [...targetNode.children, newTreeNode];
+                } else {
+                    targetNode.children = [newTreeNode];
+                }
+
+                targetNode.isSet = true;
+                targetNode.isCollapsed = false;
+
+                if (newTreeNode?.children) {
+                    newTreeNode.isCollapsed = false;
+                }
+
+                setNodeEditedFlag(draft, arrayNode, targetNode);
+                break;
+            }
+            case spiActionType.ON_REMOVE_ARRAY_ITEM: {
+                const { arrayItemToRemove } = action;
+
+                const arrayNode = PropertyInspectorModel.findPropertyTreeNodeRefRecursively(
+                    draft.propertyTreeNodes,
+                    arrayItemToRemove.path.slice(
+                        0,
+                        arrayItemToRemove.path.lastIndexOf('/')
+                    )
+                );
+
+                const originalNode = PropertyInspectorModel.findPropertyTreeNodeRefRecursively(
+                    draft.originalPropertyTreeNodes,
+                    arrayItemToRemove.path.slice(
+                        0,
+                        arrayItemToRemove.path.lastIndexOf('/')
+                    )
+                );
+
+                const childToRemoveIdx = arrayNode.children.findIndex(
+                    (el) => el.path === arrayItemToRemove.path
+                );
+
+                if (childToRemoveIdx !== -1) {
+                    arrayNode.children.splice(childToRemoveIdx, 1);
+                }
+
+                // Remove all edit status flags for map children
+                Object.keys(draft.editStatus).forEach((key) => {
+                    if (key.startsWith(arrayNode.path)) {
+                        delete draft.editStatus[key];
+                    }
+                });
+
+                setNodeEditedFlag(draft, originalNode, arrayNode);
+
+                break;
+            }
+            case spiActionType.ON_CLEAR_ARRAY: {
+                const { arrayNode } = action;
+                const arrayTreeNode = PropertyInspectorModel.findPropertyTreeNodeRefRecursively(
+                    draft.propertyTreeNodes,
+                    arrayNode.path
+                );
+
+                const originalNode = PropertyInspectorModel.findPropertyTreeNodeRefRecursively(
+                    draft.originalPropertyTreeNodes,
+                    arrayNode.path
+                );
+
+                arrayTreeNode.children.splice(0, arrayTreeNode.children.length);
+
+                // Remove all edit status flags for map children
+                Object.keys(draft.editStatus).forEach((key) => {
```

Comment:
do we have maps here? Is this in case it's an array of maps?

Reply:
I think the confusion here is in the comment. The draft.editStatus is a map of the edited paths so wee need to clear the flagged child array items from the map when clearing the root.